central/jwt/jwt.go

```diff
@@ -63,7 +92,14 @@ func create() (tokens.IssuerFactory, tokens.Validator, error) {
 		return nil, nil, errors.Wrap(err, "parsing private key")
 	}
 
-	return tokens.CreateIssuerFactoryAndValidator(issuerID, privateKey, keyID)
+	factory, validator, err := tokens.CreateIssuerFactoryAndValidator(issuerID, privateKey, keyID)
+	if err == nil {
+		validator = &m2mValidator{
+			m2m.TokenExchangerSetSingleton(roleDataStore.Singleton(), IssuerFactorySingleton()),
+			validator,
+			issuerID}
+	}
+	return factory, validator, err
```

Comment:
So if there was an error, we return `factory`, `validator` and `err` but if there was no error, we wrap the validator. Is there any reason particular reason for this? If yes, let's call it in a comment, if no, let's rewrite using a common pattern:
```
	factory, validator, err := tokens.CreateIssuerFactoryAndValidator(issuerID, privateKey, keyID)
	if err != nil {
		return nil, nil, errors.Wrap(err, "creating factory and validator")
	}

	// Re-use machine-to-machine logic to implicitly convert 3rdparty tokens... 
	// ... note the conversion will happen as part of "validation"
	wrappedValidator = &m2mValidator{
		m2m.TokenExchangerSetSingleton(roleDataStore.Singleton(), IssuerFactorySingleton()),
		validator,
		issuerID}

	return factory, wrappedValidator, err
```

Reply:
That was exactly my question: does it have to be different to justify breaking a common pattern? Looking at the code, looks like "no".

Reply:
I removed the unnecessary variable.